internal/lib/container_server.go

```diff
@@ -31,7 +31,7 @@ import (
 
 // ContainerServer implements the ImageServer
 type ContainerServer struct {
-	runtime              oci.RuntimeImpl
+	runtime              *oci.Runtime
```

Comment:
I agree with this change, but I realized that @saschagrunert introduced the change from `*Runtime` into `RuntimeImpl` some time ago with this commit 109d3709c1085bbf3ecff12b7186ac31b3655996.
@saschagrunert is there any reason why you did that in the first place? Just want to make sure we're not potentially breaking something here.

Reply:
Yes, the major reason was to have something where the behavior is mock’able, which are in golang only public interfaces.

Do you think it would be possible to extend the interface by adding a `PrivilegedWithoutHostDevices()` method?

Reply:
@saschagrunert 
>Yes, the major reason was to have something where the behavior is mock’able, which are in golang only public interfaces.
Do you think it would be possible to extend the interface by adding a PrivilegedWithoutHostDevices() method?

I see. But the interface was not created for this purpose and I'm afraid that we make code design decisions based on the capacity of mocking or not.
If we add `PrivilegedWithoutHostDevices()` to `RuntimeImpl` interface, this means `RuntimeVM` and `RuntimeOCI` implementations of this interface should also implement this new method. But that's not the goal. We want `Runtime` to remain a structure that provides the mapping into `RuntimeImpl` implementations. `Runtime` should not be seen as an implementation of `RuntimeImpl` IMO.

I think that if we really want to do proper mocking, we should add an extra abstraction layer dedicated for this, which `Runtime` would implement. But that sounds like a lot of indirection layers for the purpose of mocking.

Reply:
@saschagrunert Agree with @sboeuf there. Making this as an interface when it is not really required from an implementation point of view, kind of makes the code not very intuitive. 
This would also mean adding PrivilegedWithoutHostDevices() implementations for `RuntimeVM` and `RuntimeOCI`, when they are really not required or called.

Reply:
I think testability should be taken into consideration when designing an API, but that's not the key problem here. If we decide to leave the implementation as it currently is, then we have to remove this method which does not work any more:

https://github.com/cri-o/cri-o/blob/68070024cf96caaf252fa9b87265f8f20664bb06/internal/lib/container_server_test_inject.go#L22-L25

Afterwards, all relevant tests using this method (see `internal/lib` and `server` packages) have to be removed because they do not work any more too and I have to find another way to test these methods...maybe with a fake runtime but I'm not sure.

Reply:
@saschagrunert @sboeuf @haircommander @mrunalp 
Let me know what you think is the best way to go about this. I am going on vacation in a couple of days, and would like to get this resolved and possibly merged before that.

Reply:
To unblock this and bring it forward I'm still fine with removing the tests for now. :)

Reply:
@saschagrunert Back now, I can remove the tests for now. Can you let me know what are the tests that I need to remove?